### PR TITLE
fix: suppress duplicate action messages via suppressActionMessages/defersMessage

### PR DESCRIPTION
**Pull request by dmikalova**
Introduces a `suppressActionMessages` context flag and `defersMessage` property so that when an ability provides its own top-level message the game-action layer doesn't also emit a duplicate.

## Changes

- **CardAbility**: set `context.suppressActionMessages` when the ability declares a `message:`/`effect:`; skip game-action `effectMsg` when `action.defersMessage` is set. New `preserveActionMessages: true` opt-out for cards that want both.
- **DiscardCardAction / DiscardTopOfDeckAction / RandomDiscardAction / AllocateDamageAction**: honour `context.suppressActionMessages`
- **DiscardCardAction / DiscardTopOfDeckAction**: set `defersMessage = true`
- **RevealAction**: include owner context when revealing from hand ('{source} reveals {card} from {owner}'s hand')
- **ReturnToDeckAction**: handle empty-target edge case in effectMsg
- Remove now-redundant `chatMessage: false` from RiggedLottery, Cladogenesis (also switched to `discardTopOfDeck`), Lithomancy, HazardZerp
- **messagehelper**: strict equality (`===`) instead of `includes` for more reliable message matching
- Expand discard / reveal / archive / destroy / ward / prophecy message tests

Related to #4928

---

> [!NOTE]
> **Medium Risk**
> Changes core CardAbility and several GameActions that every card uses for logging; behavior is covered by expanded message tests but wrong deferral could hide or double-print lines in edge cases.
> 
> **Overview**
> This PR tightens **game chat logging** so abilities don’t emit duplicate or misleading lines, and updates tests/docs to match.
> 
> **Ability layer (`CardAbility`)** skips automatic game-action `effectMsg` when an action sets **`defersMessage`** (the action prints during resolution). It also avoids a bare “uses {card}” line when a **`then`** block will narrate, and ignores deferred actions when building the default message.
> 
> **Game actions** now own more of the log: **`DiscardCardAction`** / **`DiscardTopOfDeckAction`** defer upfront messaging and log discards at resolve time (including per-player “from the top of {player}'s deck”). **`RevealAction`** groups reveals by zone into **one line per hand/zone** (e.g. “reveals X and Y from player2's hand”). **`AllocateDamageAction`** gates per-target damage lines with **`chatMessage`**. **`AddTokenAction`** default text matches “place N counters on {0}” (scrap flows like Donor Vox).
> 
> **Card tweaks** turn off redundant inline chat (`chatMessage: false`) where a custom **`effect`** / **`then.message`** covers the story (e.g. Rigged Lottery, Lithomancy, Harmonic Pack), switch Cladogenesis to **`discardTopOfDeck`**, and fix Soul Lock’s **`messageArgs`**.
> 
> **Tests/helpers**: chat log extraction includes bell alerts and skips other alert types; **`toHaveAllChatMessagesBe`** uses **exact** string equality; many message specs updated for the new single-line reveals and non-duplicate discard/prophecy/ward lines.
> 
> **Docs**: new best practice to pass **card objects** in `messageArgs`/`effectArgs` for hoverable log links, not `.name` strings.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e5eb6bc18b656aca7ccf4dc31c802afbe76eafa9. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: server/game/cards/02-AoA/RiggedLottery.js [hunk 1]
```diff
@@ -6,12 +6,14 @@ class RiggedLottery extends Card {
         this.play({
             gameAction: [
                 ability.actions.discard((context) => ({
```
Comment (dmikalova) [inline server/game/cards/02-AoA/RiggedLottery.js:8]: 2026-05-12 21:24:49 player1 plays Rigged Lottery
2026-05-12 21:24:49 player1 gains an amber due to Rigged Lottery's bonus icon
2026-05-12 21:24:49 player1 discards the top 5 cards of their deck (Troll, Krump, Anger, Brammo, and Foozle) and gains 0 amber. player2 discards the top 5 cards of their deck (Gamgee, Furtive Investors, Knuckles Bolton, Bingle Bangbang, and Snufflegator) and gains 3 amber

File: server/game/cards/12-PV/Cladogenesis.js [hunk 2]
```diff
@@ -4,27 +4,14 @@ class Cladogenesis extends Card {
     // Play: Each player discards the top card of their deck and reveals their hand. Discard each card that belongs to their discarded card's house. Each player refills their hand as if it were their "draw cards" step.
     setupCardAbilities(ability) {
         this.play({
```
Comment (dmikalova) [inline server/game/cards/12-PV/Cladogenesis.js:6]: 2026-05-22 13:45:48 player1 plays Cladogenesis
2026-05-22 13:45:48 player1 gains an amber due to Cladogenesis's bonus icon
2026-05-22 13:45:48 player1 uses Cladogenesis to discard the top card of each player's deck, reveal each player's hand, and discard cards belonging to each player's discarded card's house
2026-05-22 13:45:48 player1 uses Cladogenesis to discard Anger from the top of player1's deck
2026-05-22 13:45:48 player1 uses Cladogenesis to discard Nerve Blast from the top of player2's deck
2026-05-22 13:45:48 Cladogenesis reveals Troll and Bumpsy from player1's hand
2026-05-22 13:45:48 Cladogenesis reveals Krump from player2's hand
2026-05-22 13:45:48 player1 uses Cladogenesis to discard Troll and Bumpsy
2026-05-22 13:45:48 player1 uses Cladogenesis to have each player refill their hand
2026-05-22 13:45:48 player1 draws 6 cards to refill their hand to 6 cards
2026-05-22 13:45:48 player2 draws 5 cards to refill their hand to 6 cards

File: server/game/cards/12-PV/HazardZerp.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ class HazardZerp extends Card {
                     activePromptTitle: 'Choose a card to discard',
                     location: 'hand',
                     controller: 'self',
```
Comment (dmikalova) [inline server/game/cards/12-PV/HazardZerp.js]: 2026-05-12 21:26:40 player1 discards Hazard Zerp
2026-05-12 21:26:40 player1 uses Hazard Zerp to discard Flaxia and deal 3 damage to Krump

File: server/game/cards/12-PV/Lithomancy.js [hunk 4]
```diff
@@ -18,8 +18,7 @@ class Lithomancy extends Card {
                     context.player.deck[0].hasHouse(context.player.activeHouse),
                 gameAction: [
                     ability.actions.discard((context) => ({
```
Comment (dmikalova) [inline server/game/cards/12-PV/Lithomancy.js:20]: 2026-05-12 21:28:22 player1 uses Lithomancy to reveal Dust Pixie
2026-05-12 21:28:22 player1 uses Lithomancy to discard Dust Pixie and gain 2 amber

File: server/game/GameActions/AllocateDamageAction.js [hunk 5]
```diff
@@ -40,13 +40,15 @@ class AllocateDamageAction extends GameAction {
                                         .dealDamage({ amount: amount })
                                         .getEvent(card, context)
                                 );
```
Comment (dmikalova) [inline server/game/GameActions/AllocateDamageAction.js:43]: The following suppressActionMessages changes allow the cards to suppress the default messages as needed

File: server/game/GameActions/AddTokenAction.js [hunk 6]
```diff
@@ -32,8 +32,7 @@ class AddTokenAction extends CardGameAction {
                 this.amount +
                 ' ' +
                 (this.type === 'power' ? token : this.type) +
```
Comment (dmikalova) [inline server/game/GameActions/AddTokenAction.js:34]: 2026-05-12 21:34:13 player1 discards Donor Vox
2026-05-12 21:34:13 player1 uses Donor Vox to place 2 +1 power counters on Dr. Xyloxxzlphrex

File: test/helpers/messagehelper.js [hunk 7]
```diff
@@ -41,7 +41,7 @@ function checkAllMessages(context, expectedMessages) {
             };
         }
```
Comment (dmikalova) [inline test/helpers/messagehelper.js:43]: Includes checks the strings start the same, changes it to be the strings have to be the same

File: test/server/messages/draw.spec.js [hunk 8]
```diff
@@ -62,7 +62,7 @@ describe('Draw Messages', function () {
                 'player1 uses Phase Shift to allow them to play one non-Logos card this turn',
                 'player1 draws 6 cards to refill their hand to 6 cards',
                 'player1: 0 amber (0 keys) player2: 0 amber (0 keys)',
-                'player2 does not forge a key.  They have 0 amber.  The current cost is 6 amber'
+                'player2 does not forge a key.  They have 0 amber.  The current cost is 6 amber '
```
Comment (strib) [inline test/server/messages/draw.spec.js]: Why is there an extra space now? Not a big deal but if it's easy should we trim the log lines?

**Review by strib — APPROVED** [review]

File: test/server/messages/draw.spec.js (hunk 8 above)
Comment (dmikalova) [reply test/server/messages/draw.spec.js]: A fix was removed by merge - fixed again.

**Review by strib — APPROVED** [review]
